frontend/lib/configuration/configuration.dart

```diff
@@ -1,6 +1,23 @@
 import 'package:flutter/widgets.dart';
 
-class Configuration extends InheritedWidget {
+class _Configuration extends InheritedWidget {
+  final StateConfigurationState configuration;
+
+  _Configuration({
+    required this.configuration,
+    required Widget child,
+  }) : super(child: child);
+
+  @override
+  bool updateShouldNotify(covariant _Configuration oldWidget) =>
+      configuration.mapStyleUrl != oldWidget.configuration.mapStyleUrl ||
+      configuration.graphqlUrl != oldWidget.configuration.graphqlUrl ||
+      configuration.projectId != oldWidget.configuration.projectId ||
+      configuration.showDevSettings != oldWidget.configuration.showDevSettings;
+}
+
+class Configuration extends StatefulWidget {
```

Comment:
I think it would be better not to make this a Stateful Widget.
Instead I think, we should probably simply update "the props" (React speak) of the `Configuration InheritedWidget` based on a settings change.
Currently, the "Configuration" widget is the first widget in our app widget tree. I think, we could make the SettingsProvider the first widget, and then add a stateless widget (maybe call it `ConfigurationProvider`) below it that watches the `settings.stagingEnabled` property and passes the correct values of `graphqlUrl` and `mapStyleUrl` to `Configuration`. [Currently, these values are determined in main.dart. This should be moved to the mentioned ConfigurationProvider]

That way we wouldn't have two places where the URLs are determined and we can reduce overall complexity as we don't have duplicate state in the settings and a `Stateful Configuration` widget

Reply:
reverted configuration class and changed order of providers.